addons/voice/org.openhab.voice.pollytts/src/main/java/org/openhab/voice/pollytts/internal/cloudapi/CachedPollyTTSCloudImplementation.java

```diff
@@ -0,0 +1,178 @@
+            File txtFileInCache = new File(cacheFolder, fileNameInCache + ".txt");
+            writeText(txtFileInCache, text);
+            // return from cache
+            return audioFileInCache;
+        } catch (IOException ex) {
+            logger.warn("Could not write {} to cache, return null", audioFileInCache, ex);
+            return null;
+        }
+    }
+
+    /**
+     * Gets a unique filename for a give text, by creating a MD5 hash of it. It
+     * will be preceded by the voice label.
+     *
+     * Sample: "Robert_00a2653ac5f77063bc4ea2fee87318d3"
+     */
+    String getUniqeFilenameForText(String text, String label) {
+        byte[] bytesOfMessage;
+        MessageDigest md;
+        try {
+            bytesOfMessage = text.getBytes("UTF-8");
+        } catch (UnsupportedEncodingException ex) {
+            logger.error("Could not create MD5 hash for '{}'", text, ex);
+            return null;
+        }
+        try {
+            md = MessageDigest.getInstance("MD5");
+        } catch (NoSuchAlgorithmException ex) {
+            logger.error("Could not create MD5 hash for '{}'", text, ex);
+            return null;
+        }
+        byte[] md5Hash = md.digest(bytesOfMessage);
+        BigInteger bigInt = new BigInteger(1, md5Hash);
+        String hashtext = bigInt.toString(16);
+        // Now we need to zero pad it if you actually want the full 32
+        // chars.
+        while (hashtext.length() < 32) {
+            hashtext = "0" + hashtext;
+        }
+        String fileName = label + "_" + hashtext;
+        return fileName;
+    }
+
+    // helper methods
+
+    private void copyStream(InputStream inputStream, OutputStream outputStream) throws IOException {
+        byte[] bytes = new byte[4096];
+        int read = inputStream.read(bytes, 0, 4096);
+        while (read > 0) {
+            outputStream.write(bytes, 0, read);
+            read = inputStream.read(bytes, 0, 4096);
+        }
+    }
+
+    private void writeText(File file, String text) throws IOException {
+        OutputStream outputStream = null;
+        try {
+            outputStream = new FileOutputStream(file);
+            outputStream.write(text.getBytes("UTF-8"));
+        } finally {
+            if (outputStream != null) {
+                outputStream.close();
+            }
+        }
+    }
+
+    private void updateTimeStamp(File file) throws IOException {
+        long timestamp = System.currentTimeMillis();
+        // update use date for cache management
+        file.setLastModified(timestamp);
+    }
+
+    private void purgeAgedFiles() throws IOException {
+        // just exit if expiration set to 0/disabled
+        if (PollyClientConfig.getExpireDate() == 0) {
+            return;
+        }
+        long now = new Date().getTime();
+        long diff = now - PollyClientConfig.getlastDelete();
+        // only execute ~ once every 2 days if cache called
+        long oneDay = 24 * 60 * 60 * 1000;

```

Comment:
You can use TimeUnit for calculating these values, e.g.:

`TimeUnit.DAYS.toMillis(1)`